src/services/StructuralMetadataUtils.js

```diff
@@ -68,35 +66,29 @@ export default class StructuralMetadataUtils {
    * @param {Float} duration - end time of the media file in Milliseconds
    */
   buildSMUI(allItems, duration) {
-    // Regex to match hh:mm:ss, mm:ss OR seconds(as a float)/minutes(as an int)
-    const regexHHMMSS = /^([0-9]*:[0-9]*:[0-9]*.[0-9]*)$/i;
-    const regexMMSS = /^([0-9]*:[0-9]*)$/i;
-    const regexSS = /^([0-9]*.[0-9]*)$/i;
-
     // Convert file duration to seconds
     const durationInSeconds = Math.round(duration / 10) / 100;
 
     // Convert time to HH:mm:ss.ms format to use in validation logic
     let convertToSeconds = time => {
-      if (regexMMSS.test(time)) {
-        let [minutes, seconds] = time.split(':');
-        let minutesInS = parseInt(minutes) * 60;
-        let secondsNum = seconds ? parseFloat(seconds) : 0.0;
-        return minutesInS + secondsNum;
-      } else if (regexHHMMSS.test(time)) {
-        let [seconds, minutes, hours] = time.split(':').reverse();
+      let [hours, minutes, seconds] = [];
+      // Check ss/ss.ms values
+      if (Number(time)) {
+        let secondsNum = parseFloat(time);
+        return secondsNum;
+      } else {
+        [seconds, minutes, hours] = time.split(':').reverse();
         let hoursInS = hours ? parseInt(hours) * 3600 : 0;
         let minutesInS = parseInt(minutes) * 60;
```

Comment:
In case there are no minutes, I would handle these just like hours: have a default of 0.